examples/mm/clip/test_reward_model.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pdb
+
+import numpy as np
+import torch
+import torch.multiprocessing as mp
+from omegaconf.omegaconf import OmegaConf
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import AutoModel, AutoProcessor
+
+from nemo.collections.nlp.parts.megatron_trainer_builder import MegatronTrainerBuilder
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+from nemo_aligner.data.mm.pickscore_dataset import build_train_valid_datasets
+from nemo_aligner.models.mm.stable_diffusion.image_text_rms import get_reward_model
+from nemo_aligner.utils.distributed import Timer
+
+
+@hydra_runner(config_path="conf", config_name="baseline")
+@torch.no_grad()
+def main(cfg) -> None:
+    logging.info("\n\n************** Experiment configuration ***********")
+    logging.info(f"\n{OmegaConf.to_yaml(cfg)}")
+
+    cfg.model.global_batch_size = cfg.trainer.devices * cfg.trainer.num_nodes * cfg.model.micro_batch_size
+
+    model = get_reward_model(cfg, cfg.model.micro_batch_size, cfg.model.global_batch_size).cuda()
+    model.eval()
+    batch_size = 250
+    _, val_ds, test_ds = build_train_valid_datasets(cfg.model, 0, return_test_data=True)
+    val_dl = DataLoader(val_ds, batch_size=batch_size, drop_last=False, shuffle=False, collate_fn=model.dl_collate_fn,)
+    test_dl = DataLoader(
+        test_ds, batch_size=batch_size, drop_last=False, shuffle=False, collate_fn=model.dl_collate_fn
+    )
+
+    # collect all labels here
+    all_val_probs = []
+    all_val_labels = []
+
+    # run through the val and test datasets
+    thresholds = [0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1]
```

Comment:
What is the purpose of this?

Reply:
this is how the Pickscore paper finds the optimal threshold to qualify an output as a "tie" and maximize accuracy over the validation set + report test accuracy on this parameter.